app/cli.php

```diff
@@ -6,7 +6,79 @@
 use Utopia\App;
 use Utopia\CLI\CLI;
 use Utopia\CLI\Console;
+use Utopia\Cache\Adapter\Sharding;
+use Utopia\Cache\Cache;
+use Utopia\Config\Config;
+use Utopia\Database\Database;
 use Utopia\Database\Validator\Authorization;
+use InfluxDB\Database as InfluxDatabase;
+
+function getInfluxDB(): InfluxDatabase
+{
+    global $register;
+
+    $client = $register->get('influxdb'); /** @var InfluxDB\Client $client */
+    $attempts = 0;
+    $max = 10;
+    $sleep = 1;
+
+    do { // check if telegraf database is ready
+        try {
+            $attempts++;
+            $database = $client->selectDB('telegraf');
+            if (in_array('telegraf', $client->listDatabases())) {
+                break; // leave the do-while if successful
+            }
+        } catch (\Throwable $th) {
+            Console::warning("InfluxDB not ready. Retrying connection ({$attempts})...");
+            if ($attempts >= $max) {
+                throw new \Exception('InfluxDB database not ready yet');
+            }
+            sleep($sleep);
+        }
+    } while ($attempts < $max);
+    return $database;
+}
+
+function getConsoleDB(): Database
+{
+    global $register;
+
+    $pools = $register->get('pools'); /** @var \Utopia\Pools\Group $pools */
+
+    $dbAdapter = $pools
+        ->get('console')
+        ->pop()
+        ->getResource()
+    ;
+
+    $database = new Database($dbAdapter, getCache());
+
+    $database->setNamespace('console');
+    $database->setDefaultDatabase('appwrite');
+
+    return $database;
+}
+
+function getCache(): Cache
+{
+    global $register;
+
+    $pools = $register->get('pools'); /** @var \Utopia\Pools\Group $pools */
+
+    $list = Config::getParam('pools-cache', []);
+    $adapters = [];
+
+    foreach ($list as $value) {
+        $adapters[] = $pools
+            ->get($value)
+            ->pop()
```

Comment:
Why do we need a `pop()` method ? 

What additional functionality does it offer here over simply using `get()`

Reply:
`get` is getting the `Pool` from `Group`, `pop` gets the `Connection` from `Pool`.